api/pkg/observability/deployment/deployment.go

```diff
@@ -342,11 +342,30 @@ func (c *deployer) getLabels(data *models.WorkerData) map[string]string {
 	return labels
 }
 
+func (c *deployer) getResources(data *models.WorkerData) (corev1.ResourceList, corev1.ResourceList) {
+	requests := c.resourceRequest.DeepCopy()
+	limits := c.resourceLimit.DeepCopy()
+	if data.ResourceRequest != nil {
+		if data.ResourceRequest.CPURequest != nil {
+			requests[corev1.ResourceCPU] = *data.ResourceRequest.CPURequest
+			// remove default limits
+			delete(limits, corev1.ResourceCPU)
```

Comment:
yes, by default it will use the cpu limit configured on platform-level.

Reply:
Got it. As for the CPU limit removal part, previously I was wondering if we should make it a platform-level option also, instead of removing it only when the CPU request value is set, because I felt the original behaviour seemed a little weird (if I understood it correctly):

- cpu request is configured => cpu limit removed
- cpu request not configured => default platform-level cpu limit used (either a fixed value or inexistent/no limit)

But after thinking about it, I think both are really okay so I'm fine with the current implementation 😅